frontend/src/components/Gallery/Gallery.jsx

```diff
@@ -71,6 +79,7 @@ const Gallery = ({ images, apiType }) => {
                   onLike={() => handleLike({ id, src, placeholderSrc })}
                   onExpand={() => handleShow({ src, title })}
                   onCopy={() => handleCopy({src})}
+                  onDownload={() => handleDownload({ src, downloadable })}
```

Comment:
you can import `saveAs` as `onDownload`. then pass it in here.
Like: `onDownload={() => onDownload(src, downloadable)}`